### ISSUE TITLE
Jackett wont update or change settings unless directly started as admin

### ISSUE THREAD

**Issue by klepp0906** [Jackett/Jackett#12864]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Using HTTP Client: HttpWebClient2
Using FlareSolverr: No
Using proxy: Disabled
App config/log directory: C:\ProgramData\Jackett
ThreadPool MaxThreads: 1023 workerThreads, 1000 completionPortThreads
Running in Docker: No
Jackett variant: CoreWindows
OS version: Microsoft Windows NT 10.0.19044.0 (64bit OS)
Environment version: 6.0.1 (C:\ProgramData\Jackett\)
Starting Jackett v0.20.417
```

### Description

Any time jackett attempts to update or i attempt to trigger an update it will close/crash.  This does not seem to happen when running as admin so based on that and the error message,  I can only assume its a permissions issue.

that leaves basically inquiring what my options are.  I assume its either A) autostart instead of via jackett, via something like a scheduled task or B) installing jackett elsewhere (i cant remember if this was an option via the isntaller)?  Is that it? 

### Logged Error Messages

```text
Error while deleting temp files from: C:\Users\klepp\AppData\Local\Temp\JackettUpdate-v0.20.417-637783661472382952
System.UnauthorizedAccessException: Access to the path 'clrjit.dll' is denied.
   at System.IO.FileSystem.RemoveDirectoryRecursive(String fullPath, WIN32_FIND_DATA& findData, Boolean topLevel)
   at System.IO.FileSystem.RemoveDirectory(String fullPath, Boolean recursive)
   at Jackett.Common.Services.UpdateService.CleanupTempDir() in D:\a\1\s\src\Jackett.Common\Services\UpdateService.cs:line 194

--------------------------------------------------------------------------------

Error while parsing Cardigann definition C:\ProgramData\Jackett\Definitions\ttsweb.yml
YamlDotNet.Core.YamlException: (Line: 2, Col: 3, Idx: 6) - (Line: 2, Col: 3, Idx: 6): Exception during deserialization
 ---> System.Runtime.Serialization.SerializationException: Property 'site' not found on type 'Jackett.Common.Models.IndexerDefinition'.
   at YamlDotNet.Serialization.TypeInspectors.TypeInspectorSkeleton.GetProperty(Type type, Object container, String name, Boolean ignoreUnmatched)
   at YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   --- End of inner exception stack trace ---
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.Deserializer.Deserialize(IParser parser, Type type)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](IParser parser)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](TextReader input)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](String input)
   at Jackett.Common.Services.IndexerManagerService.<>c__DisplayClass16_0.<InitCardigannIndexers>b__3(FileInfo file) in D:\a\1\s\src\Jackett.Common\Services\IndexerManagerService.cs:line 170

---------------------------------------------------------------------------------------------

Error while parsing Cardigann definition C:\ProgramData\Jackett\Definitions\legacyhd.yml
YamlDotNet.Core.YamlException: (Line: 2, Col: 3, Idx: 6) - (Line: 2, Col: 3, Idx: 6): Exception during deserialization
 ---> System.Runtime.Serialization.SerializationException: Property 'site' not found on type 'Jackett.Common.Models.IndexerDefinition'.
   at YamlDotNet.Serialization.TypeInspectors.TypeInspectorSkeleton.GetProperty(Type type, Object container, String name, Boolean ignoreUnmatched)
   at YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   --- End of inner exception stack trace ---
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.Deserializer.Deserialize(IParser parser, Type type)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](IParser parser)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](TextReader input)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](String input)
   at Jackett.Common.Services.IndexerManagerService.<>c__DisplayClass16_0.<InitCardigannIndexers>b__3(FileInfo file) in D:\a\1\s\src\Jackett.Common\Services\IndexerManagerService.cs:line 170
```

**Comment by ilike2burnthing** [comment]
Revert back to the release before last, [v0.20.413](https://github.com/Jackett/Jackett/releases/tag/v0.20.413), disable any antivirus or other security software, update Jackett and see if it crashes.

**Comment by klepp0906** [comment]
oh its been happening since i started using jackett a week or two back iirc.  i worked around it for awhile by simply disabling the auto update and updating manually but i'd prefer to stay up to date automatically of course.

i do not use any antivirus or security software.

for posterities sake i just rolled back and started jackett.  upon hitting "check for updates" it closed, never to open again.  Upon re-opening manually it did not update, but the logs show the following.

![image](https://user-images.githubusercontent.com/10392920/150585691-580e23ac-9771-40c9-ac7a-889f3e58cddb.png)

**Comment by garfield69** [comment]
If you are logged onto windows as a non-admin user then you need to run the installer with the run-as-administrator so that the service and updater can have permission to shutdown services and delete modules out of programdata and local/temp

**Comment by klepp0906** [comment]
i am admin as its my personal pc/i'm the only user.  I will try to run the installer as admin though and see how it fares.  will report back momentarily.

**Comment by klepp0906** [comment]
no bueno, if i run the installer as admin and launch it immediately from the installer menu - it will update.  however if i repeat the experiment and run the installer as admin, but dont launch it through the installer, and launch via shortcut after the fact - back to square one.

**Comment by garfield69** [comment]
I don't understand your issue.
What do you mean by saying you are running the installer via a shortcut?
Why would you need to do that?

If you installed the complete Jackett package and used the option to install as a service, then Jackett takes care of updating automatically whenever updates are available, or manually by using the Jackett dashboard check-for-updates.
Anything else is out of the design of the Jackett suite.
I'm confused.

**Comment by klepp0906** [comment]
i assume you use windows since youre referencing a service.   a shortcut is an icon that points to an executable so you can launch it as opposed to it running in perpetuity.

if you dont install as a service, installing as an application is the other option, its what happens if you uncheck the install service box when running the installer..  It's also the option i prefer which affords me to stop/start it as i see fit without needing to open services.msc and allows it to have parity with my "arrs" which are also installed in the same fashion and all update properly. 

this is the crux of the issue.  running it as a windows application does not update properly.  I'll take your word that running it as a service does.  this means the run as an application option should be removed (forced to install as a service) or fixed to update properly as a tray application.

**Comment by garfield69** [comment]
Thanks for the explanation.
I'm not qualified to resolve this issue, so I'm stepping out of this one.

**Comment by klepp0906** [comment]
sure thing.  

the workaround im going to use at current (as janky as it is) is first to have my shortcut set to run-as-admin which solves the issue if i start jackett myself after closure.   the second and larger problem is trying to be hands-off and start it automatically with windows.  for security reasons windows wont allow you start apps with admin permissions via the startup folder which is what jackett uses for autostart. 

this leaves me with either dealing with it and updating manually, forgoing autostart and starting by hand each time, or doing as ive done in the interim (under the hope that the powers that be can address it) and creating an elevated shortcut/pointing to scheduled task to allow it to start as admin on logon.

**Comment by ilike2burnthing** [comment]
When you say you do not use any AV, do you mean then that you've permanently disabled Windows Defender?

I've tried installing it not as a service and updating, it worked without issue, no need to run as admin, so not sure what issue you're facing.

Just so you're aware, you don't need to go into services.msc to stop the service, you can just right-click the Jackett tray icon and click `Stop background service`.

You can create a scheduled task to run JackettTray.exe with highest privileges on startup, no need for a shortcut with run as admin compatibility.

**Comment by klepp0906** [comment]
and you have it installed to programdata?  and youre starting it via a non admin shortcut pointing to JackettTray.exe?

Yes i disabled windows defender, as disabled as windows will allow.  no real time.  no cloud.  no tamper protection etc etc.  way too obtrusive and inconvenient but i digress. 

![image](https://user-images.githubusercontent.com/10392920/150605977-1ea7fc58-3e16-46fa-b055-ad50290af036.png)

I can disable it entirely via group policy if you think theres still some way for it to interfere but i wouldnt hold my breath. 

as for jackett as a service.  I did not know it still provided a tray icon.   THIS is news.  I assumed you simply had to create a web shortcut to access it if you installed as a service.  Though id still like to find out why not starting it as admin (or why it doesnt have admin permissions in the first place) just so i know.

**Comment by ilike2burnthing** [comment]
> and you have it installed to programdata? and youre starting it via a non admin shortcut pointing to JackettTray.exe?

Yes. Yes.

In Windows Security you could also look at:
- Device security > Core isolation
- App & browser control > all Smartscreen settings
- App & browser control > Exploit protection

To be clear, any changes to these settings should be done as a test to see if you can find a cause for your problem, and reverted if not. Any permanent changes are done at your own risk.

They really shouldn't be necessary, to my knowledge you're the only user with this issue, and I'd imagine most users have never touched those settings (well, maybe Smartscreen).

**Comment by klepp0906** [comment]
yea, i was gonna say - ill just kill defender in group policy as a test.  

could it have something to do with the fact I use brave as a browser?  It doesnt effect the others but for whatever reason, it does jackett.   something does anyways.

**Comment by ilike2burnthing** [comment]
Those settings are independent of Defender.

Your browser would have nothing to do with the update.

Your log excerpts show a wider permissions issue, as you have old indexer files which should have been deleted. Those files also could only have been installed by using a build from between 27 Feb '20 (v0.13.222) and 12 May '20 (v0.16.373).

It may be worth uninstalling Jackett, renaming/moving the Jackett folder in ProgramData, reinstalling v0.20.413, and seeing if the update works.

**Comment by klepp0906** [comment]
oh, okay well if it comes down to it then ill resort to testing by killing those settings.  seems the core isolation is already off but some of the other stuff was not, it was at whatever was default.

as for the install itself, i think you may be barking up the proper tree there.  especially if thats the case (old files).   Another thing i should have thought of/tried.

good call.  will do that after dinner and report back.  feeling optimistic :P

**Comment by klepp0906** [comment]
ah hell.  completely uninstalled, deleted what was left in programdata and reinstalled the newest version.  upon trying to change/save a setting (in this case the update to pre-release box) it wouldnt save, which is another telltale sign.  If i manually start it as admin it will.  

cant understand why its just jackett.  even after a complete clean of it. 

Error writing config file C:\ProgramData\Jackett\ServerConfig.json System.UnauthorizedAccessException: Access to the path 'C:\ProgramData\Jackett\ServerConfig.json' is denied.    at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)    at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)    at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)    at System.IO.File.WriteAllText(String path, String contents)    at Jackett.Common.Services.ConfigurationService.SaveConfig[T](T config) in D:\a\1\s\src\Jackett.Common\Services\ConfigurationService.cs:line 174

**Comment by ilike2burnthing** [comment]
Any luck with the Windows Security settings?

Failing them, if it still doesn't work, your choice is the scheduled task running at startup with highest privileges, or install as service.

**Comment by klepp0906** [comment]
honestly, i'm not even gonna mess with them.  those shouldnt need altering and everything else is fine.  i have to assume its jackett.

i found some stuff on stackoverflow referencing permissions issues with writing configs to programdata but it not only goes over my head, but it doesnt explain why all the arr's can do so.  apart from extension difference (json vs cfg i believe) theyre all in programdata and all in the root of the programs folder. 

also found this
https://forums.sonarr.tv/t/can-someone-on-here-help-me-with-a-weird-jacket-issue/25238

which sort of seems related, but he lost permissions to write to the json in the indexers folder.  mine appears to be the json in the root folder.  

I'm gonna keep digging and if push comes to shove and i get really worked up ill test disabling those obscure windows security settings, def not optimistic there as it would be effecting the slew of other softwares that run in the same fashion/are hosted locally out of that folder. 

will report back

**Comment by ilike2burnthing** [comment]
Are you running the installer as admin? If so, uninstall Jackett again and delete any leftover files, install Jackett normally, not as admin, and that should work (according to those stackoverflow threads).

But yea, failing that, just manually changing the permissions will work around it.

**Comment by klepp0906** [comment]
okay let me give it one more rip that way.  for science.  be back soonish.

**Comment by klepp0906** [comment]
nope, no dice.  installed it a few different ways & purged it each time.  I thought perhaps because i had non system spinner set as my default download location it might be related.  moved it to my system drive and still no good.  Something that is being set by jackett on install is not being set with the proper permissions/ownership.

going to try one more test and try to take ownership before running jackett the first time to see if its something on install, or something its writing/creating after running.

**Comment by klepp0906** [comment]
yep, seems its something thats written on install, not something created/written after running it the first time.

i installed but didnt open.  took ownership of folders/files.  all is right in the world.   ive done all the testing i can do but the non programmer layman in me would assume the natural order would be looking into what permissions those files the logs show are or are not created with and work backwards.. or... something.  

if theres anything else when/if any contributors take a look at this, by all means.  otherwise this is an acceptable workaround for me as opposed to creating scheduled tasks etc etc.  (assuming nothing broke by taking ownership of everything)

**Comment by ilike2burnthing** [comment]
We use fairly default Inno Setup installer settings, so we're not doing anything in regards to permissions. There is a way to do that - [Files](https://jrsoftware.org/ishelp/index.php?topic=filessection) & [Dirs](https://jrsoftware.org/ishelp/index.php?topic=dirssection) (see permissions sections) - but we don't - [Installer.iss](https://github.com/Jackett/Jackett/blob/master/Installer.iss) - nor do [Sonarr](https://github.com/Sonarr/Sonarr/blob/develop/distribution/windows/setup/sonarr.iss) or [Radarr](https://github.com/Radarr/Radarr/blob/develop/setup/radarr.iss).

If this issue was widespread, setting the permissions during installation might be a good idea, but we have a few thousand regular Windows users, almost all using auto-updates, and presumably a non-insignificant minority of them don't install it as a service, yet you're the only user to have reported this issue.

If you're now not running the installer as admin, and Sonarr and Radarr are working fine, then I really don't know why Jackett would be any different, but I can't see anything we need to change on our end.

Glad you found a way of getting it to work on your end.

**Comment by klepp0906** [comment]
Okay,  I could still be wrong but I dont believe I am.  Even if i am not, its your software that you created/work on in  your spare time as a service to everyone else and for that im envious and incredibly grateful.   I already found a workaround so this doesnt effect me beyond wanting to improve jackett and afford future users the ability to use it without needing a work around etc.

I just installed jackett on another pc.  my backup pc.   older version of windows 10, entirely different platform.   Guess what the result was?   the same.  cant change anything or auto update unless its started as admin.

I took it another step further.  I booted a brand new clean windows 10 VM.  nothing on it.  untouched.  Guess what the result was?  The same as above.

Here are the differences I can spot permissions wise.   Relevant or not goes over my head but I imagine so. 

![image](https://user-images.githubusercontent.com/10392920/150654472-b4c83ec6-48f2-4a61-bf85-b86e3a9a0024.png)

notice how all of them install with the Everyone Group?

now look here.  (forgive the smaller screenie, took it from the VM as opposed to transferring one from my other pc, and couldnt take it from my own as i already molested the directory by taking ownership)

![image](https://user-images.githubusercontent.com/10392920/150654489-e492cf1e-be46-49e1-a4e1-c98660d95981.png)

so in short, that behavior is not constrained to me or my install, and it appears to be due to the installer not adding the everyone group.

I'm unsure if you get this for closed issues, but if you do, id love to know your thoughts.

**Comment by klepp0906** [comment]
just added the everyone group with the same permissions as in the "arrs" to the jackett install on the other pc and it immediately began working.
![image](https://user-images.githubusercontent.com/10392920/150654767-e4bb0c12-053e-4b56-8b21-bd6398b3e11d.png)
![image](https://user-images.githubusercontent.com/10392920/150654770-a2755108-923a-4e12-beb0-7f38e3d3f07d.png)
![image](https://user-images.githubusercontent.com/10392920/150654776-852288d7-7a47-4520-937f-cab35d0ee0a5.png)

**Comment by ilike2burnthing** [comment]
TIL that `WDAGUtilityAccount`, the account used in Windows Sandbox, runs with elevated privileges for everything, so you can't recreate this issue in Windows Sandbox... 🤷 

I'm able to reproduce this on my actual system. Currently testing changes to the installer.

As far as I can tell the *arrs add the Everyone group as a result of
https://github.com/Sonarr/Sonarr/blob/6106362f6cac29e5690fb8bc3ab044954ffb4998/src/NzbDrone.Windows/Disk/DiskProvider.cs#L51
running Sonarr just before the installer finishes with
https://github.com/Sonarr/Sonarr/blob/6106362f6cac29e5690fb8bc3ab044954ffb4998/distribution/windows/setup/sonarr.iss#L65
so it isn't really the installer, hence me seeing no difference between theirs and ours.

**Comment by klepp0906** [comment]
LOL, i'm sure im not the only non-service user BUT i might be the only one OCD enough to not simply deal with it.  or not post an issue.   Who knows.  I know at first I simply ran as admin, disabled auto updates so it wouldnt close and then not re-open each time it tried to update, and moved on.

eventually my ocd and liking things to run as intended got to me enough to post.

either way, thanks for not taking offense.  I know persistence can go either way.  thank you again for hearing me out.  I'll wait until its addressed and I'll purge the install one last time :P

**Comment by ilike2burnthing** [comment]
Download this test build - https://dev.azure.com/Jackett/0f0cb3d0-9b6b-4a86-876d-0c2e17c18801/_apis/build/builds/5225/artifacts?artifactName=drop&api-version=7.0&%24format=zip

Extract the Windows installer and check that everything is working for you.

**Comment by klepp0906** [comment]
on it boss.  about to eat dinner real fast then will report back.

**Comment by klepp0906** [comment]
well, couldnt wait, and didnt want to keep you waiting either :P  So far so good!  will know for 100% when you push an update live but if i were a betting man.

the settings sure save properly now etc :)  thank you again for dealing with me and fixing it lol.

**Comment by ilike2burnthing** [comment]
Will be available in ~5.5hrs

**Comment by garfield69** [comment]
v0.20.428
